server/src/test/java/org/elasticsearch/cluster/ClusterStateObserverTests.java

```diff
@@ -65,4 +68,28 @@ public String toString() {
         assertTrue(listenerAdded.get());
     }
 
+    public void testWaitForState() throws Exception {
```

Comment:
Can we also test the case where the state is not immediately fulfilled? If it is not too big a hazzle with the mocking, it would seem nice to do.

Reply:
It's a big hassle I think wit the threads and everything. I think the existing coverage probably suffices here? We're using this left and right :)

Reply:
ok, let us go with that.